src/std/misc/ref.ss

```diff
@@ -0,0 +1,43 @@
+;;; -*- Gerbil -*-
+;;; (C) vyzo at hackzen.org, proksi at disroot.org
+;;; generic ref
+
+(import :gerbil/gambit
+        :std/generic)
+
+(export ref ~ :ref)
+
+(def* ref
```

Comment:
Let's define `ref-set!` as well so that we can have nice mutators with `(set! (ref obj x) val)`.

Suggested semantics:
```
(def* ref-set! 
  ((obj x val) (:set! obj x val))
  ((obj x y . rest) (apply ref-set! (:ref obj x) y rest)))
```